lib/metrics.rb

```diff
@@ -1,29 +1,51 @@
 class Metrics
-  def self.with_object(object)
-    prefix = object.class.name.underscore.tr("/", ".")
-    new(prefix)
+  include Singleton
```

Comment:
**question/quibble**: Given that this whole class holds no state whatsoever, can we make this a `Module` with `extend self` instead of a singleton class?

(Doing that would make all calls to metrics look like `Metrics.gauge(...)` which is also less cleaner IMO)

Reply:
Hmm. You are right. We need to use a Singleton only to ensure there's only one instance. The simpler call is also very tempting.

However, I'm wondering if a self-extended module will make it more like a util class than a library that's supposed to be centrally used.

@igbanam Let us know if you have any thoughts here. (Could use a tie-breaker)

Reply:
Agree with you both. I'll make this change.

Reply:
This is done